crates/biome_cli/src/cli_options.rs

```diff
@@ -100,13 +100,16 @@ pub struct CliOptions {
 
 impl CliOptions {
     /// Computes the [ConfigurationPathHint] based on the options passed by the user
-    pub(crate) fn as_configuration_path_hint(&self) -> ConfigurationPathHint {
+    pub(crate) fn as_configuration_path_hint(
+        &self,
+        working_directory: &Utf8Path,
+    ) -> ConfigurationPathHint {
         match self.config_path.as_ref() {
             None => ConfigurationPathHint::default(),
             Some(path) => {
                 let path = Utf8PathBuf::from(path);
                 let path = path.strip_prefix("./").unwrap_or(&path);
-                ConfigurationPathHint::FromUser(path.to_path_buf())
+                ConfigurationPathHint::FromUser(working_directory.join(path))
```

Comment:
This will break when the user provided an absolute path?

Reply:
Good catch, let me make a better check and I will test it too

Reply:
Works! :) [`55ef110` (#8188)](https://github.com/biomejs/biome/pull/8188/commits/55ef1100ce63032cbc13aa6570844a52f5fa406f)